crates/coresimd/tests/rotate_tests.rs

```diff
@@ -0,0 +1,49 @@
+//! rotate instruction tests
+
+#![feature(stdsimd)]
+#![feature(proc_macro)]
+#![feature(avx512_target_feature)]
+#![feature(abi_vectorcall)]
+
+extern crate stdsimd_test;
+extern crate stdsimd;
+
+use stdsimd::simd::*;
+use stdsimd_test::assert_instr;
+
+// Verify that supported hardware compiles rotates into single instructions
+
+#[inline]
+#[target_feature(enable = "avx512f")]
+#[cfg_attr(test, assert_instr(vprorvq))]
```

Comment:
The `assert_instr` and `target_feature` need to be guarded on `x86` and `x86_64`. Same for all other functions below.